collections/ansible_collections/cloudkit/service/roles/cleanup_stale_network_resources/tasks/cleanup.yaml

```
@@ -0,0 +1,256 @@
+---
+- name: Get all existing ClusterOrders from specified namespace
+  kubernetes.core.k8s_info:
+    api_version: cloudkit.openshift.io/v1alpha1
+    kind: ClusterOrder
+    namespace: "{{ cleanup_stale_network_resources_instance_name }}"
+  register: cluster_orders
+  failed_when: false
+
+- name: Initialize existing cluster names list
+  ansible.builtin.set_fact:
+    existing_cluster_names: []
+
+- name: Extract cluster names from ClusterOrders
+  ansible.builtin.set_fact:
+    existing_cluster_names: "{{ existing_cluster_names + ['clusterorder_' + item.metadata.name] }}"
+  loop: "{{ cluster_orders.resources }}"
+  when: cluster_orders.resources is defined
```

You don't need a separate task for variable initialization; you can add a `vars:` clause to the task instead:

```suggestion
- name: Extract cluster names from ClusterOrders
  ansible.builtin.set_fact:
    existing_cluster_names: "{{ existing_cluster_names + ['clusterorder_' + item.metadata.name] }}"
  loop: "{{ cluster_orders.resources }}"
  when: cluster_orders.resources is defined
  vars:
    existing_cluster_names: []
```

---

I see! When the cluster_orders.resource is [], 
```
  tasks:
    - name: Extract cluster names from ClusterOrders
      ansible.builtin.set_fact:
        existing_cluster_names: "{{ existing_cluster_names + ['clusterorder_' + item.metadata.name] }}"
      loop: "{{ cluster_orders.resources }}"
      when: cluster_orders.resources is defined
      vars:
        existing_cluster_names: []
``` 
this won't get executed, so existing_cluster_names is not defined. How should we handle this case?

---

You're already defaulting several other variables to empty lists (`[]`); you could handle `existing_cluster_names` in the same place. I think I commented somewhere else that you could do this in `vars/main.yaml` or `defaults/main.yaml`.